### PR TITLE
Box Nibbling Rework

### PR DESCRIPTION
**Pull request by NanakoAC**
Currently, a mouse gnawing open a box to get its contents generates about a screen's worth of chatspam. And the same applies to eating a box of raisins or similar large meal. I feel this isn't appropriate for the size of the creature nor the severity of the act.  In any case ive come up with a better method.

Main attraction of this PR, is that the box nibbling no longer uses visible messages, it only outputs to chat once, when it finally ruptures. In the meantime, any nibbling which doesn't break the box will instead shake it around and make a nibbling sound. Shaking code stolen from locker resist
Mice nibbling things is a quiet clicking noise, easy to ignore. so its unintrusive
**Update: I've now extended this to all animal nibbling, not only boxes, but also on food too. To cut down on chatspam from small animals eating things.**

The result of this change is that a creature eating a box or a meal is simultaneously easier than before to notice, if you care about it. And easier to ignore, if you don't care.

Also included, two bugfixes:

Re-adds the reverted console canpass behaviour that was lost a couple patches ago
adds a little robustness fix to animals, to deal with an issue where they were getting their stat set back to 0 without properly waking up. still not sure why it happens but now the icons will properly update when they do

### CODE REVIEW

**Comment by skull132** [comment]
Why not use a do_while() proc? Although, I forget if those work with simple animals.

Seems like the best solution, though!

**Comment by skull132** [comment]
@NanakoAC Feedback required on last comment.

**Comment by NanakoAC** [comment]
for the box nibbling?

Same reason we don't use a loop for eating, i guess. Roleplay.
I like the nibbling to be a conscious action, you can take it at the pace you want depending on how hungry your mouse is.
The audio effect and vibrating ensure its easy to notice if you want to, theres no need for visible messages

**Comment by Pacmandevil** [comment]
why not just delete mouses?

**Comment by NanakoAC** [comment]
Updated, conflicts resolved. In addition, the audiovisual-but-no-chatspam method is now also extended to animals eating things. Eating a box of raisins was spammy as hell

Chat messages are still generated when an animal finishes off an item of food, or when they become full and can't finish. But no more chat messages from each individual bite

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 1]
```diff
@@ -53,17 +53,32 @@
...
+/obj/item/weapon/storage/box/examine(var/mob/user)
+	..()
+	if (health < maxHealth)
+		if (health >= (maxHealth * 0.5))
+			user << span("warning", "It is slightly torn.")
+		else
+			user << span("danger", "It is full of tears and holes.")
+
+
+
```
Comment (Lohikar) [inline code/game/objects/items/weapons/storage/boxes.dm]: Please remove some of this whitespace, it doesn't contribute to readability,

File: code/_helpers/items.dm [hunk 2]
```diff
@@ -7,3 +7,10 @@
 		return 0
 	return 1
 
+
+
+/obj/proc/animate_shake()
+	var/init_px = pixel_x
+	var/shake_dir = pick(-1, 1)
+	animate(src, transform=turn(matrix(), 8*shake_dir), pixel_x=init_px + 2*shake_dir, time=1)
+	animate(transform=null, pixel_x=init_px, time=6, easing=ELASTIC_EASING)
```
Comment (Lohikar) [inline code/_helpers/items.dm]: Spaces around `=` please.

File: code/modules/modular_computers/computers/subtypes/preset_console.dm [hunk 3]
```diff
@@ -1,3 +1,14 @@
+/obj/item/modular_computer/console/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(istype(mover,/obj/item/projectile))
+		if (prob(80))
+	//Holoscreens are non solid, and the frames of the computers are thin. So projectiles will usually
+	//pass through
+			return 1
+	else if(istype(mover) && mover.checkpass(PASSTABLE))
+	//Animals can run under them, lots of empty space
+		return 1
+	return 0
```
Comment (Lohikar) [inline code/modules/modular_computers/computers/subtypes/preset_console.dm]: Do not return 0 on non-completely-dense objects. Return parent here.

File: code/game/objects/items/weapons/storage/boxes.dm (hunk 1 above)
Comment (NanakoAC) [inline code/game/objects/items/weapons/storage/boxes.dm]: done

File: code/_helpers/items.dm (hunk 2 above)
Comment (NanakoAC) [inline code/_helpers/items.dm]: done

File: code/modules/modular_computers/computers/subtypes/preset_console.dm (hunk 3 above)
Comment (NanakoAC) [inline code/modules/modular_computers/computers/subtypes/preset_console.dm]: aaaand done

**Comment by NanakoAC** [comment]
@Lohikar your points here were fixed. please update review!

**Pull request merged**